app/cafe.py

```diff
@@ -0,0 +1,19 @@
+from app.errors import NotVaccinatedError, \
+                        OutdatedVaccineError, \
+                        NotWearingMaskError
+import datetime
+
+
+class Cafe:
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    def visit_cafe(self, vstr: dict) -> str:
+        if "vaccine" not in vstr.keys() or vstr["vaccine"] == None:
+            raise NotVaccinatedError("not vaccinated")
+        elif vstr["vaccine"]["expiration_date"] < datetime.date.today():
```

Comment:
if the `expiration_date` does not exists in the dict, `KeyError` will be raised. It would be better to use `.get()` as above.
What's more, you should check if the `expiration_date`  is instance of `datetime.date` object. Otherwise `TypeError` will be raised.

Read about [isinstance() Function](https://www.w3schools.com/python/ref_func_isinstance.asp) and try it by yourself